client/auth/action-creators.ts

```diff
@@ -4,14 +4,35 @@ import { apiUrl } from '../../common/urls'
 import { SbUserId, SelfUser } from '../../common/users/sb-user'
 import { ClientSessionInfo } from '../../common/users/session'
 import type { PromisifiedAction, ReduxAction } from '../action-types'
-import type { ThunkAction } from '../dispatch-registry'
+import { openSimpleDialog } from '../dialogs/action-creators'
+import type { DispatchFunction, ThunkAction } from '../dispatch-registry'
+import i18n from '../i18n/i18next'
+import logger from '../logging/logger'
 import { abortableThunk, RequestHandlingSpec } from '../network/abortable-thunk'
 import { encodeBodyAsParams, fetchJson } from '../network/fetch'
 import { AccountUpdateSuccess, AuthChangeBegin } from './actions'
 import { getBrowserprint } from './browserprint'
 
 const typedIpc = new TypedIpcRenderer()
 
+async function changeLanguage(locale: string, dispatch: DispatchFunction<ReduxAction>) {
+  try {
+    await i18n.changeLanguage(locale)
+  } catch (error) {
+    dispatch(
+      openSimpleDialog(
+        i18n.t('auth.language.changeErrorHeader', 'Error changing the language'),
+        i18n.t(
+          'auth.language.changeErrorMessage',
+          'Something went wrong when changing the language',
+        ),
+        true,
+      ),
+    )
+    logger.error(`There was an error changing the language: ${error}`)
```

Comment:
Always use `error.stack` if it's available. Generally: `${error?.stack ?? error}`